tests/Unit/ParallelAlgorithms/Interpolation/Test_PostInterpolationCallbacks.cpp

```diff
@@ -514,8 +561,58 @@ SPECTRE_TEST_CASE(
   check_file_contents(expected_integral_a, expected_legend_a, "/SurfaceA");
   check_file_contents(expected_integral_b, expected_legend_b, "/SurfaceB");
   check_file_contents(expected_integral_c, expected_legend_c, "/SurfaceC");
+
+  // Check that the Surfaces file contains the correct surface data
```

Comment:
why not factor this into a separate function instead of a lambda? That would make the code easier to read, I think